src/ORM/Rule/ExistsIn.php

```diff
@@ -125,6 +126,51 @@ public function __invoke(EntityInterface $entity, array $options)
             }
         }
 
+        if (isset($options['parent']) && is_a($options['parent'], EntityInterface::class)) {
+            /**
+             * @var EntityInterface $parent
+             */
+            $parent = $options['parent'];
+
+            //extract entity from given parent regarding HM and HABTM associations
+            $parentEntity = null;
+            if ($isAssociation) {
+                if (TableRegistry::get($parent->getSource()) === $target->getTarget()) {
+                    //parent is of same  as target
+                    $parentEntity = $parent;
+                } else {
+                    if (!empty($parent[strtolower($target->getName())])) {
+                        $parentEntity = $parent[strtolower($target->getName())];
+                    }
+                }
+            } else {
+                $parentEntity = $parent;
+            }
+
+            if (isset($parentEntity)) {
+                //extract fields to match
+                $parent_fields = [];
+                if (is_array($parentEntity)) {
+                    //reagrd HM and HABTM associations
```

Comment:
This isn't clear to me, what are HM and reagrd?

Reply:
typo.
```php
//regard HM and HABTM associations
```

If parentEntity is an array then the association is a HasMany or HasAndBelongsToMany. Therefore there are multiple connected entities (records). If there is at least one new the old ExistsIn would fail for that and therefore for the whole entity. So the following loop extract all foreign keys to check the ids against the parent entity.